File: Bits Manipulation/Rightmost_Set_Bit.cpp
#include <bits/stdc++.h>
using namespace std;

int rightmostBit(int a)
{
    if (a == 0)
    {
        return -1;
    }

    int mask = 1;
    int count = 0;

    while ((a & mask) == 0)
    {
        mask = mask << 1;
        count++;
    }
    return count;
}

int main()
{
    int a;
    cin >> a;
    int val = rightmostBit(a);
    if (val == -1)
    {
        cout << "No Set Bits Found";
    }
    else
    {
        cout << "The Rightmost Set Bit: " << val << endl;
    }

    return 0;
}

File: Java DSA Bootcamp/src/Bitwise_Operators/RightmostSetBit.java
package Bitwise_Operators;

import java.util.Scanner;

public class RightmostSetBit {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int number = sc.nextInt();
        int mask = 1;

        System.out.println("Number: " + number);
        int ans = rightmostSetBit(number, mask);
        System.out.println("The Rightmost Set Bit is: " + ans);
    }

    private static int rightmostSetBit(int number, int mask) {
        int count = 1;
        while ((mask & number) == 0) {
            mask <<= 1;
            count++;
        }
        return count;
    }
}
